### PR TITLE
Planning Patches

### PR DESCRIPTION
**Pull request by ArtificialErmine**
Patches a few bugs in the planning service and operation loop

### CODE REVIEW

File: app/objects/c_operation.py [hunk 1]
```diff
@@ -220,24 +220,18 @@ async def _execute_atomically(self, services):
         cannot be executed. Will do a loop once through all abilities
         enumerated in adversary.
         """
+        itera = iter(range(0, len(self.adversary.atomic_ordering)))
```
Comment (khyberspache) [inline app/objects/c_operation.py:223]: defaults to 0. Also `itera` is kinda non-descript... ability_iter maybe?

Comment (khyberspache) [reply app/objects/c_operation.py:223]: `ability_iter = iter(range(len(self.adversary.atomic_ordering))`

File: app/service/planning_svc.py [hunk 2]
```diff
@@ -97,7 +98,7 @@ async def get_links(self, operation, buckets=None, agent=None, trim=True, planne
                 t = []
                 for bucket in buckets:
                     t.append([ab for ab in abilities for b in ab.buckets if b == bucket])
-                abilities = t
+                abilities = [ability for ab in t for ability in ab]
```
Comment (khyberspache) [inline app/service/planning_svc.py:101]: Okay my stupid brain is not fully processing this... is this impacted by using the index versus the ability_id? I don't see the link and it looks like we are just flattening out `t`?

Comment (ArtificialErmine) [reply app/service/planning_svc.py:101]: It's literally just flattening out t - there's some weird behavior further on if you have a list of lists at this point, so the cleanest option was just to flatten it. This is an unrelated bug patch.

File: app/objects/c_operation.py (hunk 1 above)
Comment (ArtificialErmine) [reply app/objects/c_operation.py]: Makes sense.

**Pull request merged**